### PR TITLE
Upgrade terraform-provider-minio to v1.10.0

### LINKED ISSUE #52: Upgrade terraform-provider-minio to v1.10.0
Release details: https://github.com/aminueza/terraform-provider-minio/releases/tag/v1.10.0

### PR DESCRIPTION
**Pull request by aq17**
Fixes #52

### CODE REVIEW

File: provider/go.mod [hunk 1]
```diff
@@ -6,12 +6,12 @@ replace (
...
 	github.com/minio/minio => github.com/minio/minio v0.0.0-20210409155609-0ddc4f00756b
-	github.com/minio/minio-go/v7 => github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
+	github.com/minio/minio-go/v7 => github.com/minio/minio-go/v7 v7.0.47
 	github.com/shirou/gopsutil/v3 => github.com/shirou/gopsutil/v3 v3.21.1
 )
 
 require (
-	github.com/aminueza/terraform-provider-minio v1.6.0
+	github.com/aminueza/terraform-provider-minio v1.10.0
 	github.com/pulumi/pulumi-terraform-bridge/v3 v3.28.0
```
Comment (iwahbe) [inline provider/go.mod:15]: You need bump the bridge to the latest version (v3.38.1).

File: provider/go.mod [hunk 2]
```diff
@@ -6,60 +6,73 @@ replace (
 	github.com/hashicorp/go-getter v1.5.0 => github.com/hashicorp/go-getter v1.4.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 => github.com/pulumi/terraform-plugin-sdk/v2 v2.0.0-20220725190814-23001ad6ec03
 	github.com/minio/minio => github.com/minio/minio v0.0.0-20210409155609-0ddc4f00756b
-	github.com/minio/minio-go/v7 => github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
+	github.com/minio/minio-go/v7 => github.com/minio/minio-go/v7 v7.0.47
```
Comment (jazzyfresh) [inline provider/go.mod:9]: did you run the `go mod tidy` step? i thought this was supposed to mark it with a timestamp after the tag

**Review by jazzyfresh — APPROVED** [review]
LGTM

v7.0.11 -> v7.0.47 refreshes a provider that was a year stale. Great work!

File: provider/go.mod (hunk 2 above)
Comment (iwahbe) [reply provider/go.mod:9]: You should only get a timestamp if your upstream isn't released properly or you're targeting a sha. This looks right to me.

File: provider/go.mod (hunk 1 above)
Comment (jazzyfresh) [reply provider/go.mod]: let's include this in the upgrade guide instructions

**Pull request merged**